provider/pkg/provider/environment.go

```diff
@@ -59,6 +63,15 @@ func ToPulumiServiceEnvironmentInput(properties *structpb.Struct) (*PulumiServic
 
 	input := PulumiServiceEnvironmentInput{}
 	input.OrgName = inputMap["organization"].StringValue()
+
+	// Set project to "default" if not in input
+	input.ProjectName = defaultProject
+
+	inputProject, ok := inputMap["project"]
+	if ok && inputProject.StringValue() != "" {
```

Comment:
There are helper functions to make this check neater, like so:
`if inputMap["project"].HasValue() && inputMap["project"].IsString() {`